### PR TITLE
feat: Add attendee custom form option

### LINKED ISSUE #3199: Step 2 : Allow addition of custom fields in the attendee-form
Parent Issue : #2956 

This issue is focused on implementing a basic custom question addition in the attendee forms just like event-brite.

Proposed addition in the custom-fields model:
- Question Prompt : To let organisers add details about the custom question
- IsCustomQuestion : To separate the default fields from these custom fields.

### PR DESCRIPTION
**Pull request by iamareebjamal**
Fixes #3199

### CODE REVIEW

**Comment by iamareebjamal** [comment]
It isn't even complete yet :man_shrugging:

**Comment by sparsh-99** [comment]
Means, your approach going in right direction....

**Comment by iamareebjamal** [comment]
There is no approach. I have literally just changed let to const

File: app/components/forms/wizard/attendee-step.js [hunk 1]
```diff
@@ -1,6 +1,10 @@
 import Component from '@ember/component';
 import FormMixin from 'open-event-frontend/mixins/form';
 
+function getIdentifier(name, fields) {
```
Comment (niranjan94) [inline app/components/forms/wizard/attendee-step.js:4]: Codacy found an issue: ['fields' is defined but never used.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5730106)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity decreasing per file
==============================
+ app/models/custom-form.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5730106&bid=18745667)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity decreasing per file
==============================
+ app/models/custom-form.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5730106&bid=18745667)

File: app/components/forms/wizard/attendee-step.js (hunk 1 above)
Comment (niranjan94) [inline app/components/forms/wizard/attendee-step.js:4]: Codacy found an issue: ['fields' is defined but never used.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5730106)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- app/components/forms/wizard/attendee-step.js  1
- app/utils/text.ts  1
         

Complexity decreasing per file
==============================
+ app/models/custom-form.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5730106&bid=18745667)

File: app/utils/text.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export function slugify(string: string, sep: string = '-'): string {
+    return string.toLowerCase()
+        .replace(/[^\w\s-]/g, '') // remove non-word [a-z0-9_], non-whitespace, non-hyphen characters
```
Comment (niranjan94) [inline app/utils/text.ts:3]: Codacy found an issue: [Expected indentation of 4 spaces but found 8.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5730106)

File: app/utils/text.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+        .replace(/[\s_-]+/g, sep) // swap any length of whitespace, underscore, hyphen characters with a single -
+        .replace(/^-+|-+$/g, ''); // remove leading, trailing -
```
Comment (niranjan94) [inline app/utils/text.ts:5]: Codacy found an issue: [Expected indentation of 4 spaces but found 8.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5730106)

File: app/utils/text.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export function slugify(string: string, sep: string = '-'): string {
+    return string.toLowerCase()
+        .replace(/[^\w\s-]/g, '') // remove non-word [a-z0-9_], non-whitespace, non-hyphen characters
+        .replace(/[\s_-]+/g, sep) // swap any length of whitespace, underscore, hyphen characters with a single -
```
Comment (niranjan94) [inline app/utils/text.ts:4]: Codacy found an issue: [Expected indentation of 4 spaces but found 8.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5730106)

File: app/utils/text.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export function slugify(string: string, sep: string = '-'): string {
+    return string.toLowerCase()
```
Comment (niranjan94) [inline app/utils/text.ts:2]: Codacy found an issue: [Expected indentation of 2 spaces but found 4.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5730106)

File: tests/unit/utils/text-test.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { slugify} from 'open-event-frontend/utils/text';
```
Comment (niranjan94) [inline tests/unit/utils/text-test.ts:1]: Codacy found an issue: [A space is required before '}'.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5730106)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- app/utils/text.ts  1
- app/components/forms/wizard/attendee-step.js  1
         

Complexity decreasing per file
==============================
+ app/models/custom-form.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5730106&bid=18745667)

File: app/utils/text.ts (hunk 4 above)
Comment (niranjan94) [inline app/utils/text.ts:4]: Codacy found an issue: [Expected indentation of 4 spaces but found 8.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5730106)

File: app/utils/text.ts (hunk 5 above)
Comment (niranjan94) [inline app/utils/text.ts:2]: Codacy found an issue: [Expected indentation of 2 spaces but found 4.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5730106)

File: app/utils/text.ts (hunk 2 above)
Comment (niranjan94) [inline app/utils/text.ts:3]: Codacy found an issue: [Expected indentation of 4 spaces but found 8.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5730106)

File: app/utils/text.ts (hunk 3 above)
Comment (niranjan94) [inline app/utils/text.ts:5]: Codacy found an issue: [Expected indentation of 4 spaces but found 8.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=5730106)

**Pull request merged**